src/ReflectionFileNamespace.php

```diff
@@ -131,12 +132,14 @@ public function getConstant($constantName)
     /**
      * Returns a list of defined constants in the namespace
      *
+     * @param bool $withDefinedConstants Include constants defined via "define(...)" in results.
+     *
      * @return array
      */
-    public function getConstants()
+    public function getConstants($withDefinedConstants = false)
```

Comment:
You should separate caches with global constants by storing them in two different arrays in order to make it working for both cases and subsequent calls.